assets/js/components/notifications/dashboard-auth-alert.js

```diff
@@ -28,18 +28,26 @@ import Notification from '../notifications/notification';
 
 const DashboardAuthAlert = () => {
 	const { admin: { connectURL } } = global.googlesitekit;
-	const { currentAdminPage } = global.googlesitekit.admin;
-	const product = currentAdminPage
-		.replace( /googlesitekit|module|-/g, ' ' )
-		.replace( /(^\w{1})|(\s{1}\w{1})/g, ( match ) => match.toUpperCase() )
-		.trim();
-
+	const { unsatisfiedScopes } = global.googlesitekit.setup;
+	let message = '';
+	const missingScopes = [ ...new Set( unsatisfiedScopes.map( ( item ) => item.match( /https:\/\/www.googleapis.com\/auth\/([a-z]+)/ )[ 1 ] ) ) ];
+	const moduleNames = missingScopes.map( ( module ) => module.charAt( 0 ).toUpperCase() + module.slice( 1 ) );
+	if ( 1 < moduleNames.length ) {
+		/* translators: %1$s: Product name */
+		message = sprintf( __( 'Site Kit can’t access the some relevant data because you haven’t granted all API scopes requested during setup. To use Site Kit, you’ll need to redo the setup for: %1$s – make sure to approve all API scopes at the authentication stage.', 'google-site-kit' ), moduleNames.join( ', ' ) );
+	} else if ( moduleNames.length === 1 ) {
+		/* translators: %1$s: Product name */
+		message = sprintf( __( 'Site Kit can’t access the relevant data from %1$s because you haven’t granted all API scopes requested during setup. To use Site Kit, you’ll need to redo the setup for %1$s – make sure to approve all API scopes at the authentication stage.', 'google-site-kit' ), moduleNames[ 0 ] );
+	} else {
+		// Generic error message here.
+		message = __( 'Site Kit can’t access the relevant data from because you haven’t granted all API scopes requested during setup. Please redo setup.', 'google-site-kit' );
```

Comment:
I realize we're pending final verbiage here but the "from because" here is a bit off 😄